lightkurve/correctors/cbvcorrector.py

```diff
@@ -1,29 +1,735 @@
+        sector=None, channel=None, module=None, output=None, camera=None,
+        CCD=None, cbv_type='SingleScale', band=None, cbv_indices='ALL'):
+
+    """ Searches the `public data archive at MAST <https://archive.stsci.edu>`
+    for Kepler, K2 or TESS cotrending basis vectors.
+
+    This function fetches the Cotrending Basis Vectors FITS HDU for the desired
+    mission and channel or CCD, etc... and then extracts the requested basis
+    vectors
+
+    For Kepler/K2, the FITS files contain all channels in a single file per
+    quarter/campaing. But for TESS, each CCD CBVs are stored in a seperate FITS
+    file. So, the retrieval process is slightly different betweem the missions.
+
+    Depending on which mission you are searching for different named parameters
+    must be passed as discussed below. If additional named arguments are passed
+    then an error is thrown
+
+    Parameters
+    ----------
+    mission     : str, list of str
+                    'Kepler', 'K2', or 'TESS'.
+    quarter or campaign or sector : int, list of ints
+                    Kepler Quarter, K2 Campaign, or TESS Sector number.
+    channel or module and output : int
+                    Kepler/K2  requested channel or module and output
+                    Must provide either channel, or module and outout
+    camera and CCD : int, list of ints
+                    TESS camera and CCD
+    cbv_type    : str
+                    'SingleScale' or 'MultiScale' or 'Spike'
+                    For Kepler/K2 only single-scale is available
+    band        : int
+                    Multi-scale band number
+    cbv_indices : int array
+                    List of CBVs extracted from FITS file, 1-Based {'ALL' => extract all}
+
+    Returns
+    -------
+    result : :class:`CBVFile` object
+        Object detailing the data products found.
+
+    Examples
+    --------
+    This example will read in the CBVs for Kepler quarter 8, 
+    and then extract the first 8 CBVs for module.output 16.4
+
+        >>> cbvs = get_cbvs(mission='Kepler', quarter=8, module=16, output=4,   # doctest: +SKIP
+        >>>     cbv_indices=np.arange(1,9))                                     # doctest: +SKIP
+
+    This example will read in the CBVs for TESS Sector 10 Camera.CCD 2.4
+    and then extract the first 6 CBVs of multi-scale band 2
+
+        >>> cbvs = search_cbvs(mission='TESS', sector=10, camera=2, CCD=4, # doctest: +SKIP
+        >>>     cbv_type='MultiScale', band=2, cbv_indices=np.arange(1,7)) # doctest: +SKIP
+    """
+
+    #***
+    # Validate inputs 
+    # Make sure only the appropriate arguments are passed
+    # TODO: figure out a more elegant way to do this
+    if (mission == 'Kepler'):
+        assert  isinstance(quarter, int), 'quarter must be passed for Kepler mission'
+        # All these inputs are invalid
+        assert  campaign is None, 'campaign must not be passed for Kepler mission'
```

Comment:
This is a convention decision. It's your code, so if you would prefer two custom functions versus a single, more versatile function then I can do that. I suppose it would just throw an error if the TESS getter was called for a Kepler light curve. However, I think in general the average user would never call get_cbvs directly anyway, they would construct a corrector object which would retrieve the CBVs automatically.